Ported/combat-bees/Assets/Scripts/Systems/MovementSystem.cs

```diff
@@ -22,22 +21,54 @@ public void Execute(in ArchetypeChunk chunk, int unfilteredChunkIndex, bool useE
         var transforms = chunk.GetNativeArray(TransformHandle);
         var velocities = chunk.GetNativeArray(VelocityHandle);
 
-        var gravityDt = TimeStep * Gravity; 
+        var gravityDt = TimeStep * Gravity;
         for (int i = 0; i < chunk.Count; ++i)
         {
             var tmComp = transforms[i];
             var velComp = velocities[i];
 
             var velNew = velComp.Value;
-            velNew.y += gravityDt; 
             var pos = tmComp.Value.Position;
+            float k_DeepImpactVel = 0.5f;
+            float k_Restitution = 0.4f;
+            float k_FrictionDamping = 0.3f;
+
+            velNew.y += gravityDt;
             pos += TimeStep * velNew;
-            // bounce off lower bottom
-            pos.y *= Convert.ToSingle(pos.y > 0);
+            float notOnGround = Convert.ToSingle(pos.y > 0);
+
+            if (notOnGround != 1.0f && velNew.y < -k_DeepImpactVel) // deep impact
+            {
+                velNew.y = -velNew.y * k_Restitution;
+                velNew.x *= k_FrictionDamping;
+                velNew.z *= k_FrictionDamping;
+            }
+            else
+            {
+                velNew.y *= notOnGround;
+                velNew.x *= notOnGround;
+                velNew.z *= notOnGround;
+            }
+
+            // stop at lower bottom
+            pos.y *= notOnGround;
 
             tmComp.Value.Position = pos;
             velComp.Value = velNew;
 
+            // derive heading of velocity
+            var heading = math.atan2(velNew.x, velNew.z);
+            var headingRot = Quaternion.AngleAxis(math.degrees(heading), Vector3.up);
```

Comment:
For up/down/left/right you can use math.up, math.left, etc.  UnityEngine math types are still structs, so they work fine in Burst, but it is preferred to use Unity.Mathematics as much as possible  in Burst compiled methods (this is not always possible, especially with APIs in UnityEngine)

Reply:
Good to know. Thanks.